recipe/conda_build_config.yaml

```
@@ -0,0 +1,2 @@
+c_stdlib_version:
+  - "2.28"
```

I know this PR is mainly for testing (thanks :pray), but just to make sure: unless `nccl` _requires_ a newer glibc to be compiled (if so, how did this recipe work until now?), you shouldn't increase the `c_stdlib_version` concurrently with the image bump. To avoid https://github.com/conda-forge/conda-smithy/issues/2159 you just need to use a selector
```suggestion
c_stdlib_version:  # [linux]
  - "2.28"         # [linux]
```

---

Picked NCCL as it was a nice simple test case of building for CUDA 11/12

Agree this shouldn't be needed actually (as it works fine on GLIBC 2.17)

Indeed adding a selector should fix it; however, it shouldn't be needed either

In any case am more concerned by `docker pull` not working atm